File: src/gui-tesing-library/Models/GTSystemVersion.cs
using System;

namespace gui_tesing_library.Models;

public class GTSystemVersion
{
    private readonly long _Build;
    private readonly long _Major;
    private readonly long _Minor;
    private readonly string _OsType;

    public GTSystemVersion(GTSystemVersion_CS gTSystemVersion_CS)
    {
        _OsType = gTSystemVersion_CS.GetOsType();
        _Minor = gTSystemVersion_CS.GetMinor();
        _Major = gTSystemVersion_CS.GetMajor();
        _Build = gTSystemVersion_CS.GetBuild();
    }

    public GTSystemVersion(OperatingSystem oSVersion)
    {
        _OsType = oSVersion.Platform.ToString();
        _Major = oSVersion.Version.Major;
        _Minor = oSVersion.Version.Minor;
        _Build = oSVersion.Version.Build;
    }

    public string VersionString => $"{_OsType} {_Major}.{_Minor} Build: {_Build}";
}

File: src/GtLibrary/GTSystemVersion.cpp
#include "pch.h"
#include "GTSystemVersion.h"

GTSystemVersion::GTSystemVersion(const std::string& osType, long major, long minor, long build)

    : _OsType(osType), major(major), minor(minor), build(build) {
};
GTSystemVersion::GTSystemVersion(const GTSystemVersion& other)
    : _OsType(other._OsType), major(other.major), minor(other.minor), build(other.build) {
};
std::string GTSystemVersion::GetVersionString() const {
	return _OsType + " " + std::to_string(major) + "." + std::to_string(minor) + " Build: " + std::to_string(build);
}
